aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -495,7 +612,25 @@ impl AggregatorData {
             .aggregator_snapshots
             .get(&id)
             .expect("AggregatorSnapshot doesn't exist");
-        snapshot.value
+        match snapshot.state {
```

Comment:
I think we want read_snapshot() to be equivalent to read(), as in - doing the expensive read.